test/imath/ref/out.xml

```diff
@@ -0,0 +1,517 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```

Comment:
This is going to sound annoying. I didn't think of it before. But maybe json would be nicer than xml. To avoid the '&lt;' and '&gt' in the names. Just thought of it now. Not a big deal though.

Reply:
Yeah I agree that the string escaping is ugly as hell. I don't mind json either. I'll try it out see what it looks like.

Reply:
also would mean the size and align parameters can actually be ints in the file rather than strings.

Reply:
Only downside to that would be being limited to 31 bits for ids as ints. That’s /probably/ enough, but famous last words and all that...

Reply:
Ah I didn't think of that.

Reply:
JSON output is much nicer. I’m going to get free function matching working today then make a PR for that and continue working in there as the readability makes development much easier at this stage.

I think I’ll also add a basic generator that just rebuilds the C++ from the JSON as that should make debugging a lot easier too.

Reply:
Ah awesome. My plan for the c output refactor is to just transform the ast then dump it with a c output instead of json, so there will be cross over.